model-optimizer/extensions/front/kaldi/split_lstm_memoryoffset.py

```diff
@@ -0,0 +1,62 @@
+"""
+ Copyright (C) 2018-2020 Intel Corporation
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+import networkx as nx
+
+from mo.front.common.replacement import FrontReplacementSubgraph
+from mo.graph.graph import Graph
+from mo.ops.memoryoffset import MemoryOffset
+from mo.ops.result import Result
+from mo.utils.error import Error
+
+
+class SplitLstmMemoryOffset(FrontReplacementSubgraph):
+    """
+    Splits MemoryOffsets in LSTM blocks into 2 parts. These parts then will be converted to ReadValue and Assign.
```

Comment:
Why we say that this is an LSTM block? Could it be another type of recurrent network? Probably we should use word "recurrent" here...

Reply:
You are right, in the general case there also can be other "recurrent" blocks not only LSTM and this transformation will also trigger it that cases. For the general case yes it will be reasonable to name `recurrent` instead of `LSTM` but currently the only recurrent block we have in Kaldi models is LSTM. To minimize efforts I prefer to leave the name as it is now. 

Do you insist to rename It?

Reply:
Fair enough. Renamed LSTM -> recurrent. And mentioned in comments that at the moment we have only LSTM blocks with MemOffsets. Please review.